### PR TITLE
[APP-2545] Copy Firebase analytics from GH

### PR DESCRIPTION
**Pull request by G0retZ**
**What does this PR do?**

   Copies and adapts Firebase analytics from GH. 

   Found and fixed some issues. Also had to refactor some code for more convenient analytics integration.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] By commits

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APP-2545](https://aptoide.atlassian.net/browse/APP-2545)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-2545](https://aptoide.atlassian.net/browse/APP-2545)

**What are the relevant PRs?**

  PRs related to this one: [#531](https://github.com/Aptoide/aptoide-client-dt/pull/531)  [#2](https://github.com/Catappult/appcoins-payments-android/pull/2)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-2545]: https://aptoide.atlassian.net/browse/APP-2545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-2545]: https://aptoide.atlassian.net/browse/APP-2545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app-games/src/main/java/com/aptoide/android/aptoidegames/analytics/presentation/AnalyticsContext.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/analytics/presentation/AnalyticsContext.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/AptoideApplication.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/AptoideApplication.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/categories/presentation/CategoriesView.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/categories/presentation/CategoriesView.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/installer/presentation/InstallViewStates.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/installer/presentation/InstallViewStates.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/LandscapePaymentView.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/LandscapePaymentView.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/MoreBundleView.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/MoreBundleView.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/notifications/NotificationsPermissionRequester.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/notifications/NotificationsPermissionRequester.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/PortraitPaymentView.kt
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/PortraitPaymentView.kt:1]: Apply code style

File: app-games/src/main/java/com/aptoide/android/aptoidegames/network/AptoideGetHeaders.kt [hunk 1]
```diff
@@ -54,5 +54,5 @@ class AptoideGetHeaders @Inject constructor(
     return "${locale.language}-${locale.country}, ${locale.language};q=0.9, *;q=0.7"
   }
 
-  override val channel: String = "aptoide-games"
+  override val channel: String = "ANDROID"
```
Comment (GuiFPires) [inline app-games/src/main/java/com/aptoide/android/aptoidegames/network/AptoideGetHeaders.kt:57]: leave it as "aptoide-games"

**Review by GuiFPires — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/network/AptoideGetHeaders.kt (hunk 1 above)
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/network/AptoideGetHeaders.kt]: Fixed

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/PortraitPaymentView.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/PortraitPaymentView.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/notifications/NotificationsPermissionRequester.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/notifications/NotificationsPermissionRequester.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/MoreBundleView.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/MoreBundleView.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/LandscapePaymentView.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/payment_methods/LandscapePaymentView.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/installer/presentation/InstallViewStates.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/installer/presentation/InstallViewStates.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/categories/presentation/CategoriesView.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/categories/presentation/CategoriesView.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/AptoideApplication.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/AptoideApplication.kt:1]: Applied

File: app-games/src/main/java/com/aptoide/android/aptoidegames/analytics/presentation/AnalyticsContext.kt
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/analytics/presentation/AnalyticsContext.kt:1]: Applied

**Review by GuiFPires — APPROVED** [review]

**Review by GuiFPires — APPROVED** [review]

**Review by ricardo-frango — CHANGES_REQUESTED** [review]

File: app-games/src/main/java/com/aptoide/android/aptoidegames/analytics/presentation/SwipeListener.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    RealSwipeListener(interactionSource)
+  }
+}
+
+@Composable
+fun RealSwipeListener(
+  interactionSource: InteractionSource,
+) {
+  val genericAnalytics = rememberGenericAnalytics()
+  val (numberOfSwipes, setNumberOfSwipes) = remember { mutableStateOf(0) }
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply app-games/src/main/java/com/aptoide/android/aptoidegames/analytics/presentation/SwipeListener.kt]: Fixed

**Review by GuiFPires — APPROVED** [review]

**Pull request merged**